packages/cli/src/platform/auth/login.ts

```diff
@@ -27,111 +28,107 @@ export class Login implements Command {
-                searchParams.delete('user')
-                location.pathname += '/success'
-                const nextSearchParams = new URLSearchParams({
-                  ...Object.fromEntries(searchParams.entries()),
-                  email: user.email,
-                })
-                location.search = nextSearchParams.toString()
-                resolve({ token, user })
-              } else {
-                location.pathname += '/error'
-                location.searchParams.set('error', 'Invalid user')
-                reject(new Error('Invalid user'))
-              }
+              location.pathname += '/error'
+              location.searchParams.set('error', 'Invalid user')
+              reject(new Error('Invalid user'))
             }
-
-            res.statusCode = 302
-            res.setHeader('location', location.href)
-            res.end()
-          })
-          server.once('error', reject)
-        }),
-        open(authSigninUrl.href),
-      ])
-
-      const result = await writeAuthConfig({ token: authResult.token })
-      if (isError(result)) throw result
-
-      return successMessage(`Authentication successful for ${authResult.user.email}`)
-    } catch (error) {
-      throw new Error(`Authentication failed: ${isError(error) ? error.message : ''}`)
+          }
+
+          res.statusCode = 302
+          res.setHeader('location', location.href)
+          res.end()
+        })
+        server.once('error', reject)
+      }),
+      open(loginUrl.href),
+    ])
+      .then((results) => results[0])
+      .catch(unknownToError)
+
+    if (isError(callbackResult)) throw new Error(`Authentication failed: ${callbackResult.message}`) // prettier-ignore
+
+    {
+      const writeResult = await credentialsFile.save({ token: callbackResult.token })
+      if (isError(writeResult)) throw new Error('Writing credentials to disk failed', { cause: writeResult })
     }
+
+    return successMessage(`Authentication successful for ${callbackResult.user.email}`)
   }
 }
 
-const generateAuthSigninUrl = async (params: { connection: string; redirectTo: string }) => {
-  const cliSignature = await checkpoint.getSignature().catch((e) => {
-    debug(`await checkpoint.getSignature() failed silently with ${e}`)
-    return null
-  })
-
-  const state = {
-    client: `${PRISMA_CLI_NAME}@${PRISMA_CLI_VERSION}`,
-    // will be `null` if it throws during retrieval
-    // will be a UUIDv4 when successful
-    signature: cliSignature,
+const createLoginUrl = async (params: { connection: string; redirectTo: string }) => {
+  const userAgent = await getUserAgent()
+  const state: State = {
+    client: userAgent,
     ...params,
   }
-  const stateEncoded = Buffer.from(JSON.stringify(state), `utf-8`).toString(`base64`)
-  const url = new URL('/auth/cli', platformConsoleUrl)
+  const stateEncoded = encodeState(state)
+  const url = new URL('/auth/cli', consoleUrl)
   url.searchParams.set('state', stateEncoded)
   return url
 }
```

Comment:
@prisma/platform This will require some changes in PDP to correctly read the signature. Though, actually, I am not sure we use the signature at all on login. @sampolahtinen ?

Reply:
> not sure we use the signature at all on login. @sampolahtinen ?

Previously, we used signature only on signups.

Reply:
We agreed in standup today to use the UA going forward.